joomla/cassiopeia
# Metismenu - Fix dropdown toggle for parent menu items with links

Pull Request for Issue # .

### Summary of Changes

When using a menu item of a type which has a link as a parent for a dropdown, the metismenu js toggles the aria-expanded properties and expands and collapses the dropdown then clicking on the menu item's link and not when clicking on the dropdown toggle.

This PR corrects that by using the `triggerElement` parameter when creating the metismenu.

With help of the same parameter it was also possible to change the dropdown toggle from a linke with role="button" to a real button element.

Thanks to @bembelimen and @hans2103 for help with it.

### Testing Instructions

1. Checkout the [development branch](https://github.com/joomla/cassiopeia/tree/development) in your Cassipeia local git clone and if necessary pull the latest changes.

2. If never done before:
- Run `composer install` and `npm ci`.
- Install Blog Sample data.

3. In the "Main Menu Blog", change the type of those menu items which have sub-menu items from "Menu Heading" to something else which has a link, e.g. "Featured Articles".

4. Check the dropdown toggle of the metismenu in frontend (= horizontal menu in the header area).
Result: See section "Actual result" below.

5. Apply the chages of this PR.

6. Run `npm run build:css` or `npm ci` to compile css.

7. Check again the dropdown toggle of the metismenu in frontend.
Result: See section "Expected result" below.

8. Inspect the menu's markup with browser tools.
Result: See section "Expected result" below.

### Expected result

The dropdown toggles when clicking the dropdown toggle.

When clicking on the menu item's link, the corresponding page is shown.

A `<button>` element is used for the dropdown toggle.

### Actual result

The dropdown toggles when clicking on the menu item's link and not when clicking on the dropdown toggle.

### Documentation Changes Required

None.
